### ISSUE TITLE
CSS doesn't seem to work when casting Lovelace page to my Google Hub

### ISSUE THREAD

**Issue by flyize** [thomasloven/lovelace-card-mod#129]
My Home Assistant version: 2021.6.0

My lovelace configuration method (GUI or yaml): GUI

What I am doing: Casting a Lovelace tab

What I expected to happen: Additional CSS to be rendered

What happened instead: Nothing

**Minimal** steps to reproduce:

```yaml
# The least amount of code possible to reproduce my error
        card_mod:
          style: |
            .name {
            font-size: 30px;
            }
            div {
            font-size: 20px;
            }
# End of code
```

Error messages from the browser console:

---

**By putting an X in the boxes ([]) below, I indicate that I:**

- [x] Understand that this is a channel for reporting bugs, not a support forum (https://community.home-assistant.io/).

- [x] Have made sure I am using the latest version of the plugin.

- [x] Have followed the troubleshooting steps of the "Common Problems" section of https://github.com/thomasloven/hass-config/wiki/Lovelace-Plugins.

- [x] Understand that leaving one or more boxes unticked or failure to follow the template above may increase the time required to handle my bug-report, or cause it to be closed without further action.

**Comment by UBWH** [comment]
+1 on this

CSS via card-mod working great when browser dispalys the view.

But - casting does not display the same CSS mods.

**Comment by greberg** [comment]
Do you still have problem using card-mod with casting to Google Nest Hub? I have problem with it as well. From that everything worked as expected to now, not working. All other custom components seems to work. Could it maybe be that not right module 3.1.2 is loaded by the casting. Will try to factory reset the hub and see if that helps. Have you tried that @UBWH ?

**Comment by greberg** [comment]
> 

Yes the casting has been flakey. Especially when Google moved to Fuchia os. They have a strange implementation that for a lot of people mean that you need to re-cast every 10 minutes. So casting works ok with this and the CSS has been working flawless until for 2022.3 release. Casting still works but the CSS seems to not work. Not sure if it is a cache problem here or something else. Will try factory reset and maybe deploy casting locally on my network instead through Nabu Casa

**Comment by UBWH** [comment]
I am casting on local LAN (not via Nabu Casa). Yes - have https URL with LetsEncrypt SSL cert.
Both the Hub & HA server on same LAN.

CSS works VERY rarely.

My casts last for hours (generally) but not days.

**Comment by greberg** [comment]
Factory reset didn't work. Will report back when running local instead

**Comment by flyize** [comment]
In my case, this was actually completely self inflicted. At some point I added some copy of 'cast' to my custom_components folder. Once I removed that, everything started working again.

**Comment by greberg** [comment]
So I have setup everything locally now so I can inspect the Google Hub Device. This shows something about if Uncaught (in promise) TypeError: t.refresh is not a function at HTMLElement.o.update (/hacsfiles/lovelace-card-mod/card-mod.js:6) 

I'm not an expert within javascript so I need someone to have a look at this, if this actually is a problem. But what chrome://inspect shows it seems to be with the if statement and the t.refresh:
`if (o.update && !o.update.cm_patched) {
                    const e = o.update;
                    o.update = function(o) {
                        e.bind(this)(o),
                        t.refresh(),
                        this.updateComplete.then((()=>{
                            t.refresh()
                        }
                        ))
                    }
                    ,
                    o.update.cm_patched = !0
                }`

**Comment by greberg** [comment]
Have manage to get it to work by installing version 3.0.12 and now it works. As soon as I deploy later version it stops working and create tons of issues pointing to t.refresh.

**Comment by suneworld** [comment]
> Have manage to get it to work by installing version 3.0.12 and now it works. As soon as I deploy later version it stops working and create tons of issues pointing to t.refresh.

Fantastic. Worked for me! Thank you so much!

**Comment by markus99** [comment]
Updated to 3.1.4 after the April Fools incident and I'm no longer getting the CSS via cast.  Seemed like it worked 4 out of 5 times previously (and a re-cast would fix before).  Now only getting non-styled view.  Using Lets Encrtypt SSL via DuckDNS.  Casting to a Lenovo 10" Smart Display.

**Comment by flyize** [comment]
Just to reiterate what my problem was. Please make sure that you don't have a custom_components/cast folder under you HA config folder.

**Comment by markus99** [comment]
Worth noting, I just downgraded to 3.0.12 and it's working correctly again.  Appears something changed b/w 3.0.12 and 3.1.4 that Google Cast doesn't like.

**Comment by kleinerhobbit** [comment]
I have expiriensed the same issue.
Often no elements with Card-Mod items.
After downgrade to 3.0.12 everything is working smooth.
I am casting with catt to Google Nest Hub 2.

**Comment by jdiegmueller** [comment]
Same issue here, running HA 2022.5.2 currently.  

Using the built-in Google Cast service (cast.show_lovelace_view service).

Casting to a Chromecast Built-In baked in to a Sony Bravia XR-65X90J.

**Comment by jtonk** [comment]
card-mod recently updated to 3.1.5, but the problem is still there. Version 3.1.1 works fine though...

**Comment by Mas-Tool** [comment]
> So I have setup everything locally now so I can inspect the Google Hub Device. This shows something about if Uncaught (in promise) TypeError: t.refresh is not a function at HTMLElement.o.update (/hacsfiles/lovelace-card-mod/card-mod.js:6)
> 
> I'm not an expert within javascript so I need someone to have a look at this, if this actually is a problem. But what chrome://inspect shows it seems to be with the if statement and the t.refresh: `if (o.update && !o.update.cm_patched) { const e = o.update; o.update = function(o) { e.bind(this)(o), t.refresh(), this.updateComplete.then((()=>{ t.refresh() } )) } , o.update.cm_patched = !0 }`

Hi @greberg, I was wondering if you could share how you got were able to debug the page on the google nest hub.
I would like to try to solve this issue but I don't see the nest hub on the chrome inspect.

**Comment by greberg** [comment]
@Mas-Tool Your nest hub should appear in chrome://inspect if you are in the same network. If no then I'm not really sure really why. For me the nest hub appear in the list when I run inspect

**Comment by Kenny3231** [comment]
Hello, I still have the incident with version 3.2.0

**Comment by Techgeekster** [comment]
I am also experiencing this issue, any updates on resolving this?

**Comment by arnocl** [comment]
Same here, my nest hub refuses to show the css mods.

**Comment by jurgenweber** [comment]
Yeah, I am just casting to a chrome cast onto my TV. The CSS worked once, I am sure.. but then it stopped. Any solutions?

**Comment by matthewclaussen** [comment]
I have this same problem with casting on my setup. Any information I can provide?

**Comment by maxi1134** [comment]
Anyone ever found a solution?

**Comment by walkercz** [comment]
Hello,
same problem here on chromecast 3rd gen no customisation applied when casting the dashboard
thank you for your time

**Comment by jurgenweber** [comment]
apparently not.. annoying.

**Comment by audacity707** [comment]
Having the same issue.

**Comment by Thomsen-79** [comment]
same issue here.. not possible to downgrade card mod to anything before 3.1.4

**Comment by panda7789** [comment]
Same issue here

**Comment by tonnguyen** [comment]
> same issue here.. not possible to downgrade card mod to anything before 3.1.4

You can use whatever version you want by manually installing instead of using HACS, follow [this guide](https://github.com/thomasloven/hass-config/wiki/Lovelace-Plugins).

When selecting the .js file to download, choose the version you want from this: https://github.com/thomasloven/lovelace-card-mod/tags

**Comment by Kippy74** [comment]
I am having this issue as well..
Any news about this?

**Comment by demiandenker** [comment]
I found a workaround that has been working reliably for me for over a week now. 

I cast a morning dashboard every day at 7:30am using an automation that:
- turns off any previous casting
- casts a morning dashboard view with just a Markdown card saying "Loading"
- waits 10 secs
- casts the morning dashboard view with the actual dashboard to show that includes the cards using card_mod.

This is my automation script:

```
alias: Show Morning Dashboard
description: ""
triggers:
  - trigger: time
    at: "07:30:00"
conditions: []
actions:
  - action: media_player.turn_off
    metadata: {}
    data: {}
    target:
      device_id: [google_nest_hub]
  - action: cast.show_lovelace_view
    data:
      entity_id: media_player.living_room_display
      dashboard_path: morning-dashboard
      view_path: "0"
  - delay:
      hours: 0
      minutes: 0
      seconds: 10
      milliseconds: 0
  - action: cast.show_lovelace_view
    data:
      entity_id: media_player.living_room_display
      dashboard_path: morning-dashboard
      view_path: "1"
mode: single
```

I'm using:
card_mod v3.4.3
HA v2024.10.4
Google Nest Hub 2

**Comment by eddiewebb** [comment]
@demiandenker solution worked for me. Important the first step is `device`, not entity, and I had to increase my delay slightly because of connection slowness. (otherwise was stuck on loading screen)
